sql-cli/src/sql_cli/dag_generator.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from networkx import DiGraph, depth_first_search, is_directed_acyclic_graph
+from sql_directory_parser import SqlFile
+
+
+class SqlFilesDAG:
+    """A DAG of sql files i.e. used for finding the right order to execute the sql files in."""
+
+    @staticmethod
+    def build(sql_files: set[SqlFile]) -> list[SqlFile]:
+        """
+        Build a directed graph from a list of sql files.
+
+        :params sql_files: The sql files to use for DAG generation.
+
+        :returns: a list of sql files sorted by least dependencies first
+            so they can be called sequentially in python code.
+        """
+        graph = DiGraph()
+
+        for sql_file in sql_files:
+            graph.add_node(sql_file.get_variable_name())
+            for parameter in sql_file.get_parameters():
+                graph.add_edge(sql_file.get_variable_name(), parameter)
+
+        if not is_directed_acyclic_graph(graph):
+            raise Exception(
+                "Could not generate DAG! Please make sure your sql files do not have circular dependencies."
```

Comment:
Would it be possible for us to detail where the circular dependency is, to helping the user to troubleshoot the issue?

Reply:
Resolved in cb8f6de